core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -73,17 +70,11 @@ object BattleDamage {
             val civResources = civInfo.getCivResourcesByName()
             for (resource in combatant.unit.baseUnit.getResourceRequirements().keys)
                 if (civResources[resource]!! < 0 && !civInfo.isBarbarian())
-                    modifiers["Missing resource"] = -25
+                    modifiers["Missing resource"] = -25  //todo ModConstants
 
-
-            val nearbyCivUnits = combatant.unit.getTile().getTilesInDistance(2)
-                .flatMap { it.getUnits() }.filter { it.civInfo == combatant.unit.civInfo }
-            if (nearbyCivUnits.any { it.hasUnique("Bonus for units in 2 tile radius 15%") }) {
-                val greatGeneralModifier =
-                    if (combatant.unit.civInfo.hasUnique(UniqueType.GreatGeneralProvidesDoubleCombatBonus)) 30 else 15
-
-                modifiers["Great General"] = greatGeneralModifier
-            }
+            val greatGeneralModifier = GreatGeneralImplementation.getGreatGeneralBonus(combatant.unit)
+            if (greatGeneralModifier.second != 0)
+                modifiers[greatGeneralModifier.first] = greatGeneralModifier.second
```

Comment:
I'd prefer extracting this variable immedatialy, e.g.,
```kotlin
val (unitName, modifier) = GreatGeneralImplementation.getGreatGeneralBonus([...])
```
But this is a minor point, feel free to ignore

Reply:
My words, if I remember correctly and didn't forget to hit the green button :+1: And 'datialy' sounds tasty. I'll leave the box and visit kitchen imme- wait - dates - ly. Or was that dates :cherries:  from Italy :it: ?